applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/flow/FlowTests.kt

```diff
@@ -731,6 +740,109 @@ class FlowTests {
         assertThat(result.flowStatus).isEqualTo(RPC_FLOW_STATUS_SUCCESS)
     }
 
+    // TODO CORE-7939 For now this flow WILL succeed, however, this will require modifications once the ledger has been
+    //  finalised. Specifically, create a state before trying to spend it.
+    @Test
+    fun `Notary - Non-validating plugin executes successfully and returns signatures`() {
+        val requestID =
+            startRpcFlow(
+                bobHoldingId,
+                emptyMap(),
+                "net.cordapp.testing.testflows.NonValidatingNotaryTestFlow"
+            )
+        val result = awaitRpcFlowFinished(bobHoldingId, requestID)
+        assertThat(result.flowStatus).isEqualTo(RPC_FLOW_STATUS_SUCCESS)
+        assertThat(result.flowResult)
+            .isEqualTo("Received 1 signatures from the notary, plugin ran successfully.")
+    }
+
+    @Test
+    fun `Notary - Non-validating plugin returns error when time window invalid`() {
+        val requestID =
+            startRpcFlow(
+                bobHoldingId,
+                mapOf(
+                    "timeWindowUpperBoundOffsetMs" to "-1000",
+                    "timeWindowLowerBoundOffsetMs" to "-2000"
+                ),
+                "net.cordapp.testing.testflows.NonValidatingNotaryTestFlow"
+            )
+        val result = awaitRpcFlowFinished(bobHoldingId, requestID)
+
+        assertThat(result.flowStatus).isEqualTo(RPC_FLOW_STATUS_FAILED)
+        assertThat(result.flowError?.message).contains("Unable to notarise transaction")
+        assertThat(result.flowError?.message).contains("NotaryErrorTimeWindowOutOfBounds")
+    }
+
+    // TODO CORE-7939 For now it's impossible to test this scenario as the `LedgerTransaction` will always return an
+    //  empty list of input state and refs (no back-chain resolution)
+    @Test
+    @Disabled
+    fun `Notary - Non-validating plugin returns error when double spend occurs`() {
```

Comment:
```suggestion
    fun `Notary - Non-validating plugin returns error when using reference state that is spent in same tx`() {
```

Avoids confusion - I'd assume a "double spend" is trying to spend the same input state in two different txs.